### PR TITLE
RFC - Deprecate duplicate methods in Router

### PR DESCRIPTION
**Pull request by markstory**
All of these methods have nicer/better cousins on `RouteBuilder`. By deprecating these methods we can remove them for 4.x. This is a possibly a contentious change as I suspect a few people are using these methods. I _really_ wanted to remove `Router::connect()` as well but I feel that might be too big of a change.

One of the reasons I want to remove these methods is they create broken use cases when combined with scopes. For example:

``` php
Router::scope('/api', function ($routes) {
  Router::mapResources('Articles');
});
```

Does not do what you want. There are many other ways to get into these bad states, and I'd prefer to not have them.

### CODE REVIEW

**Comment by lorenzo** [comment]
This is going to take a lot of people by surprise, but good stuff!

**Comment by markstory** [comment]
@lorenzo Should we not do this now? I really wanted to remove `Router::connect()` but I felt that might need more warning. If you think this is going to anger lots of folks we can hold off until a later 3.x release.

File: src/Routing/Router.php [hunk 1]
```diff
@@ -219,9 +219,14 @@ public static function connect($route, $defaults = [], $options = [])
      *   shifted into the passed arguments. As well as supplying patterns for routing parameters.
      * @return void
      * @see \Cake\Routing\RouteBuilder::redirect()
+     * @deprected 3.3.0 Use Router::scope() and RouteBuilder::redirect() instead.
```
Comment (thinkingmedia) [inline src/Routing/Router.php]: should be ``deprecated``
missing the **a**

Comment (markstory) [inline src/Routing/Router.php]: Thank you.

**Comment by thinkingmedia** [comment]
I'm not a fan of run-time errors for deprecated methods. The phpDoc notation is enough to flag the method in the IDE (I use PHPStorm). Anyone not using an IDE with phpDoc support is out of luck I guess.

**Comment by markstory** [comment]
@thinkingmedia Runtime errors ensure that people can't say there wasn't warning. Would you recommend we do doc tags now and add runtime errors closer to 4.0? For example 3.5 or 3.6?

**Comment by lorenzo** [comment]
Let's tag first, remove any reference to those methods in the docs, and in 3.4 we start triggering errors

**Comment by jhli** [comment]
:+1: for notices at some point (I don't care when).

**Comment by markstory** [comment]
@lorenzo I'm good with that.

**Pull request merged**